utils/default-helpers/src/main/scala/pl/touk/nussknacker/engine/util/functions/numeric.scala

```diff
@@ -57,7 +59,23 @@ trait numeric extends MathUtils with HideToString {
   @Documentation(description = "Parse string to number")
   @GenericType(typingFunction = classOf[ToNumberTypingFunction])
   def toNumber(@ParamName("stringOrNumber") stringOrNumber: Any): java.lang.Number = stringOrNumber match {
-    case s: CharSequence     => new java.math.BigDecimal(s.toString)
+    case s: CharSequence =>
+      val ss = s.toString
+
+      val tries: List[Try[java.lang.Number]] = List(
```

Comment:
I would add a comment that we pick the narrowest type as possible to reduce the amount of memory and computations overheads ,